tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassLoaderAdapter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.junit4.runners;
+
+import static org.apache.commons.beanutils.MethodUtils.invokeMethod;
+
+/**
+ * Just an utility class to have in one place the calls using reflection due to {@link org.mule.runtime.module.artifact.classloader.ArtifactClassLoader}
```

Comment:
This javadoc is not quite the same to the explanation you gave me about why this class is needed. Try to improve it